addons/website_sale_picking/models/delivery_carrier.py

```diff
@@ -0,0 +1,33 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    # Onsite delivery means the client comes to a physical store to get the products himself.
+    delivery_type = fields.Selection(selection_add=[
+        ('onsite', 'On Site')
+    ], ondelete={'onsite': lambda record: record.write({'delivery_type': 'fixed'})})
```

Comment:
'fixed' is the default so we can just set default here.

```suggestion
    ], ondelete={'onsite': 'set default')
```